StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1649FileNameMustMatchTypeName.cs

```diff
@@ -84,22 +84,22 @@ public void HandleSyntaxTreeAction(SyntaxTreeAnalysisContext context)
                 }
 
                 var fileName = Path.GetFileName(context.Tree.FilePath);
-                string expectedFileName;
+                string[] expectedFileNames;
```

Comment:
:question: Since this is in an analyzer (as opposed to a code fix), can we implement this without the additional array and closure allocations?